### PR TITLE
Fix race condition problems

### PR DESCRIPTION
**Pull request by denis-tingajkin**
## Motivation and Context
https://github.com/networkservicemesh/networkservicemesh/issues/1725

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by denis-tingajkin** [comment]
unit tests passed with `-race` flag https://circleci.com/gh/networkservicemesh/networkservicemesh/113065?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link

File: controlplane/pkg/services/client_connection_manager.go [hunk 1]
```diff
@@ -375,34 +381,50 @@ func (m *ClientConnectionManager) UpdateFromInitialState(xcons []*crossconnect.C
...
 func (m *ClientConnectionManager) CleanupDeletedConnections() {
 	// Iterate over connections to cleanup any orphaned.
-	for k, c := range m.clientConnections {
-		if c.deleted && time.Since(c.deleteTime) > deletedConnectionLifetime {
-			// remove connection if hold for a long time already.
-			delete(m.clientConnections, k)
+	idSlice := []string{}
+	m.clientConnections.Range(func(k, v interface{}) bool {
+		if c, ok := v.(*managedClientConnection); ok {
+			c.Lock()
```
Comment (lobkovilya) [inline controlplane/pkg/services/client_connection_manager.go]: I suppose `sync.RWMutex` is more suitable for that case

File: test/cloudtest/pkg/utils/executor.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+var once sync.Once
+var instance Executor
+
+//DefaultExecutor return default executer
+func DefaultExecutor() Executor {
+	once.Do(
+		func() {
+			e := &executor{executables: make(chan operation, 100)}
+			go e.run()
+			instance = e
```
Comment (lobkovilya) [inline test/cloudtest/pkg/utils/executor.go]: What purpose to use global executor?

Comment (denis-tingajkin) [reply test/cloudtest/pkg/utils/executor.go]: thanks, fixed

**Pull request merged**